### DISCUSSION TITLE
Schematic

### DISCUSSION THREAD

**Discussion by RichardPar** [meatpiHQ/wican-fw#57] [General]
Is there a schematic for the ODB-2 version ? (pinouts for what connects to the ESP32)   I would like to write some firmware for 

Richard

**Comment by thefunkygibbon** [comment]
i too would like to create my own given the cost of the wican inc shipping and that I have a bunch of obd2 and esp32 devices kicking about doing nothing. 

Don't suppose you found this did you?  I assume you could just use this firmware on your own esp32 but the pins will be defined already

**Reply by meatpiHQ** [reply]
Here you go:

CAN RX -> GPIO3
CAN TX -> GPIO0
VBAT ADC -> GPIO4
Blue LED -> GPIO7
Green LED -> GPIO8
Yellow LED -> GPIO9
CAN transceiver Standby -> GPIO6

**Reply by osklar098** [reply]
Hi,
Are the pinouts the same for WiCAN-USB? 
/Oskar

**Reply by meatpiHQ** [reply]
Yes it's the same.

**Reply by Freddy-0** [reply]
Hi @thefunkygibbon,

Did you get this project working with a regular esp32?

I am in a similar situation. 
I have some esp32 (non s3 or c3) boards laying around and trying to compile this project for one, but getting stuck. If you got this working, i would like to open a new discussion about this.

**Reply by Telemachus1977** [reply]
Using Expressif IDE, I changed the target to ESP32. The only file that he's changed is hw_config.h to set the GPIO pins that you would like to use. I needed to modify the sdkconfig to turn on BLE and some other components based on errors.  Set flash to 4mb and it compiled and flashed just fine.

**Comment by fowi4hjte** [comment]
And here is for the current WiCAN-PRO adapter: https://github.com/meatpiHQ/wican-fw/blob/main/sch/wican_obd_pro_sch_v151.pdf

**Reply by ankur-paan** [reply]
i can see one TJA1042 in the schematics, but there are 4 on the actual wican pro, how esp32 is communicating via all 4 at once? i think esp32 can communicate with only 1 chip at a time.
